.cirrus.yml

```diff
@@ -1,192 +1,61 @@
-    - ./configure $DEFAULT_CONFIG_OPTS
-  make_distcheck_script:
-    - make -j$(nproc) -s distcheck DISTCHECK_CONFIGURE_FLAGS="${DEFAULT_CONFIG_OPTS}"
-  always:
-    make_check_artifacts:
-      path: "**/*.log"
-
-###
-# Default toolchain and build flags used in deb packages, on a range of Debian
-# and Ubuntu releases (+ Debian/unstable)
-# Most should succeed, and PRs shouldn't break them.
-#
-debian_default_toolchain_task:
-  matrix:
-    - container:
-        image: collectd/ci:debian12
-      allow_failures: false
-      env:
-        CONTAINER_IMAGE: debian12
-    - container:
-        image: collectd/ci:ubuntu23.10
-      allow_failures: false
-      env:
-        CONTAINER_IMAGE: ubuntu23.10
-    - container:
-        image: collectd/ci:debian_unstable
-      allow_failures: true
-      skip_notifications: true
-      only_if: $CIRRUS_BRANCH == 'main'
-      env:
-        CONTAINER_IMAGE: debian_unstable
+    # build using clang with a collection of strict build flags, will most
+    # probably always fail
+  allow_failures: true
+  skip_notifications: true
+  git_submodule_script:
+    - git submodule init -- opentelemetry-proto
+    - git submodule update -- opentelemetry-proto
   configure_script:
     - ./build.sh
-    - gcc --version
+    - clang --version
     - >
-      ./configure CC=gcc $DEFAULT_CONFIG_OPTS
-      CFLAGS="$(dpkg-buildflags --get CFLAGS)"
-      CPPLAGS="$(dpkg-buildflags --get CPPFLAGS)"
-      LDFLAGS="$(dpkg-buildflags --get LDFLAGS)"
-  build_script:
-    - make -j$(nproc) -sk
-  tests_script:
-    - VALGRIND_OPTS="--errors-for-leak-kinds=definite" make -j$(nproc) -sk check || (cat ./test-suite.log && false)
-    - |
-      if [[ -e ".expected-plugins/$CONTAINER_IMAGE" ]]; then
-        /checks/check-built-plugins.sh ".expected-plugins/$CONTAINER_IMAGE"
-      else
-        /checks/check-built-plugins.sh
-      fi
-  always:
-    make_check_artifacts:
-      path: "**/*.log"
-
-###
-# Default toolchain and build flags used in RPM packages, on a range of RedHat
-# and Fedora releases (+ Fedora/rawhide)
-# Most should succeed, and PRs shouldn't break them.
-#
-redhat_default_toolchain_task:
-  matrix:
-    - container:
-        image: collectd/ci:centos9
-      allow_failures: false
-      env:
-        CONTAINER_IMAGE: centos9
-    - container:
-        image: collectd/ci:fedora39
-      allow_failures: false
-      env:
-        CONTAINER_IMAGE: fedora39
-    - container:
-        image: collectd/ci:fedora_main
-      allow_failures: true
-      skip_notifications: true
-      only_if: $CIRRUS_BRANCH == 'main'
-      env:
-        CONTAINER_IMAGE: fedora_main
-  configure_script:
-    - ./build.sh
-    - gcc --version
-    - ./configure CC=gcc $DEFAULT_CONFIG_OPTS CFLAGS="$(rpm --eval '%optflags')"
+      ./configure CC=clang CXX=clang++
+      $DEFAULT_CONFIG_OPTS
+      CFLAGS='-gdwarf-4 -Wall
+      -Wno-error
+      -Wextra
+      -Wformat=2
+      -Wformat-security
+      -Wformat-nonliteral
```

Comment:
At least with GCC, latter 2 are implied by the first one (according to `man gcc`).